lib/settingsManager.ts

```diff
@@ -0,0 +1,68 @@
+import { DeepPartial } from "utility-types";
```

Comment:
This file is really hard to read and  disables half of the typescript checkers. I would really want to have specific unit tests for this.
Less stuff with files but just calling the different functions with example objects. Also as a specification what this is supposed to do.

Reply:
Yeah, that was your function ;)